### DISCUSSION TITLE
Formik with Listbox and default value (codesandbox  provided)

### DISCUSSION THREAD

**Discussion by nicolas-besnard** [tailwindlabs/headlessui#1695] [Help]
Hi :) 

I have this [codesandbox][1] that reproduces my problem. When I have values defined by my form (with `Formik`), I'm able to select the value that is already present in my initial value.
![Screenshot-2022-07-18-20-45-07-wKjiB6uY](https://user-images.githubusercontent.com/1261670/179581111-dbae67a4-f5e6-40cc-a698-b389a04ba396.gif)

I'm not sure what am doing wrong here. Any help would be really appreciated 

[1]: https://codesandbox.io/s/charming-khorana-depu4s?file=/src/App.tsx

**Comment by ShaneYu** [comment]
I am getting the same issue at this very moment with the latest insiders release too.

It seems like the initially selected values bound via `value` prop on the combobox are not being seem as "selected" as they don't have a ticket icon next to them when first rendered. When selecting one of the already selected options it seems to treat it separately as a different value and as such you end up with duplicates selected.

I've also tried using other props such as `defaultValue` without success to work around this issue.

**Comment by ShaneYu** [comment]
Okay, I've done some experimentation and I believe the issue is that the combobox  is matching selected items by javascript reference(s) when binding objects as values. I.E. the objects used to populate the available options often are not the same objects "selected" and so do now match.

Ideally I think the combobox needs to support a method of checking selected/matched values using a specified property on the objects; not sure they nicest way to deal with this.

In my case, to work around this issue, I changed the value binding to map from object to an array of "ids", changed the option values to use a id property rather than the entire object and then changed the onChange to map from an id array to a items array using the items used to populate the combobo box; hope that makes sense.

_Short version is, binding data in converts to id array, binding out on change converts from ids to items._

---

I've forked and updated your codesandbox to demonstrate how I worked around the issue and gets it working in your scenario.
https://codesandbox.io/s/gracious-jasper-wklh3q?file=/src/App.tsx

**Comment by jlefebvre1997** [comment]
In case someone comes across this post as I did, the Listbox component allows you to specify a field to use to identify which value is selected or not : the `by` prop 
https://headlessui.com/react/listbox#binding-objects-as-values
